### PR TITLE
[EEEC] New object type Event Consumption Model

### PR DESCRIPTION
**Pull request by huber-nicolas**
We should put an AFF Example Object for EEEC into package TEST_AFF_EXAMPLES.
The interface  /iwxbe/if_aff_eeec_v1 should be renamed if possible.
A test object for integration tests is missing.

### CODE REVIEW

File: file-formats/eeec/README.md [hunk 1]
```diff
@@ -0,0 +1,6 @@
+# EEEC File Format
+This file format is used for the ADT Editor for EEEC
```
Comment (huber-nicolas) [inline file-formats/eeec/README.md:2]:
```suggestion
This file format is used by the ADT editor for Event Consumers (EEEC).
```

File: file-formats/eeec/type/zif_aff_eeec_v1.intf.abap [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+    ty_type TYPE string.
+
+  TYPES:
+    "! <p class="shorttext synchronized" lang="en">Event Type</p>
+    BEGIN OF ty_s_event_type,
+      "! <p class="shorttext synchronized" lang="en">Type</p>
+      "! Type
+      type  TYPE ty_type,
+      "! <p class="shorttext synchronized" lang="en">Entity Name</p>
+      "! Entity Name
```
Comment (wurzka) [inline file-formats/eeec/type/zif_aff_eeec_v1.intf.abap:16]:
```suggestion
      "! Entity name
```

File: file-formats/eeec/type/zif_aff_eeec_v1.intf.abap [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+    "! <p class="shorttext">Event Consumption Model</p>
+    BEGIN OF ty_main,
+      "! $required
+      format_version TYPE zif_aff_types_v1=>ty_format_version,
+      "! <p class="shorttext">Header</p>
+      "! Header
+      "! $required
+      header           TYPE zif_aff_types_v1=>ty_header_60,
+      "! <p class="shorttext synchronized" lang="en">Consumer Class</p>
+      "! Consumer Class
```
Comment (wurzka) [inline file-formats/eeec/type/zif_aff_eeec_v1.intf.abap:31]:
```suggestion
      "! Consumer class
```

Comment (wurzka) [inline file-formats/eeec/type/zif_aff_eeec_v1.intf.abap:31]:
```suggestion
      "! Consumer class
```

**Comment by wurzka** [comment]
ToDo: Add example object

File: file-formats/eeec/type/zif_aff_eeec_v1.intf.abap [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+
+  TYPES:
+    "! <p class="shorttext">Event Consumption Model</p>
+    BEGIN OF ty_main,
+      "! $required
+      format_version TYPE zif_aff_types_v1=>ty_format_version,
+      "! <p class="shorttext">Header</p>
+      "! Header
+      "! $required
+      header           TYPE zif_aff_types_v1=>ty_header_60,
```
Comment (wurzka) [inline file-formats/eeec/type/zif_aff_eeec_v1.intf.abap:29]:
```suggestion
      header           TYPE zif_aff_types_v1=>ty_header_60_cloud,
```

File: file-formats/eeec/eeec-v1.json [hunk 5]
```diff
@@ -0,0 +1,104 @@
...
+          "maxLength": 2,
+          "pattern": "^[a-z]+$"
+        },
+        "abapLanguageVersion": {
+          "title": "ABAP Language Version",
+          "description": "ABAP language version",
+          "type": "string",
+          "enum": [
+            "standard",
+            "keyUser",
```
Comment (wurzka) [inline file-formats/eeec/eeec-v1.json:39]:
```suggestion

```

File: file-formats/eeec/eeec-v1.json [hunk 6]
```diff
@@ -0,0 +1,104 @@
...
+            "cloudDevelopment"
+          ],
+          "enumTitles": [
+            "Standard",
+            "ABAP for Key Users",
```
Comment (wurzka) [inline file-formats/eeec/eeec-v1.json:44]:
```suggestion

```

File: file-formats/eeec/eeec-v1.json [hunk 7]
```diff
@@ -0,0 +1,104 @@
...
+            "ABAP Cloud Development"
+          ],
+          "enumDescriptions": [
+            "Standard",
+            "ABAP for key user extensibility",
```
Comment (wurzka) [inline file-formats/eeec/eeec-v1.json:49]:
```suggestion

```

**Review by wurzka — COMMENTED** [review]
EEEC is only for cloud development. So 'keyUser' is no valid value for ABAP Language Version

File: file-formats/eeec/eeec-v1.json [hunk 8]
```diff
@@ -0,0 +1,104 @@
...
+        "type": "object",
+        "properties": {
+          "type": {
+            "title": "Type",
+            "description": "Type",
+            "type": "string"
+          },
+          "entityName": {
+            "title": "Entity Name",
+            "description": "Entity Name",
```
Comment (wurzka) [inline file-formats/eeec/eeec-v1.json:88]:
```suggestion
            "description": "Entity name",
```

File: file-formats/eeec/README.md [hunk 9]
```diff
@@ -0,0 +1,6 @@
+# EEEC File Format
+This file format is used by the ADT editor for Event Consumers (EEEC).
```
Comment (schneidermic0) [inline file-formats/eeec/README.md:2]: We have got feedback to not list the use case of the object type here. Therefore, I suggest to remove it.
```suggestion
```

File: file-formats/eeec/type/zif_aff_eeec_v1.intf.abap [hunk 10]
```diff
@@ -0,0 +1,42 @@
...
+  TYPES:
+    "! <p class="shorttext synchronized" lang="en">Event Type</p>
+    BEGIN OF ty_s_event_type,
+      "! <p class="shorttext synchronized" lang="en">Type</p>
+      "! Type
+      type  TYPE ty_type,
+      "! <p class="shorttext synchronized" lang="en">Entity Name</p>
+      "! Entity name
+      entity_name TYPE zif_aff_types_v1=>ty_object_name_30,
+    END OF ty_s_event_type,
```
Comment (schneidermic0) [inline file-formats/eeec/type/zif_aff_eeec_v1.intf.abap:18]: Do we need `$required` fields here?

**Review by schneidermic0 — CHANGES_REQUESTED** [review]
I have one very small change request. And one question. Otherwise it looks fine to me.

File: file-formats/eeec/type/zif_aff_eeec_v1.intf.abap (hunk 10 above)
Comment (wurzka) [reply file-formats/eeec/type/zif_aff_eeec_v1.intf.abap:18]: I will check with Raphael

Comment (wurzka) [reply file-formats/eeec/type/zif_aff_eeec_v1.intf.abap]: 'type' and 'entity_name' are required, now

**Review by schneidermic0 — APPROVED** [review]
Cool :)

LGTM

**Comment by wurzka** [comment]
I created issue #342 for the example object so that I can close this pull request.

**Pull request merged**